ios/browser/ui/webui/brave_wallet/brave_wallet_page_ui.mm

```diff
@@ -0,0 +1,188 @@
+#import "ios/web/public/web_state.h"
+#include "ios/web/public/webui/url_data_source_ios.h"
+#import "ios/web/public/webui/web_ui_ios.h"
+#import "ios/web/public/webui/web_ui_ios_data_source.h"
+#import "ios/web/public/webui/web_ui_ios_message_handler.h"
+#include "ui/base/accelerators/accelerator.h"
+#include "ui/base/webui/web_ui_util.h"
+
+namespace {
+
+web::WebUIIOSDataSource* CreateAndAddWebUIDataSource(
+    web::WebUIIOS* web_ui,
+    const std::string& name,
+    const webui::ResourcePath* resource_map,
+    std::size_t resource_map_size,
+    int html_resource_id) {
+  web::WebUIIOSDataSource* source = web::WebUIIOSDataSource::Create(name);
+  web::WebUIIOSDataSource::Add(ChromeBrowserState::FromWebUIIOS(web_ui),
+                               source);
+  source->UseStringsJs();
+
+  // Add required resources.
+  source->AddResourcePaths(base::make_span(resource_map, resource_map_size));
+  source->SetDefaultResource(html_resource_id);
+  return source;
+}
+
+}  // namespace
+
+BraveWalletPageUI::BraveWalletPageUI(web::WebUIIOS* web_ui, const GURL& url)
+    : web::WebUIIOSController(web_ui, url.host()) {
+  web::WebUIIOSDataSource* source;
+  if (url.path() == "/swap" || url.path() == "/send") {
+    source = CreateAndAddWebUIDataSource(web_ui, url.host(),
+                                         kBraveWalletIosContainerGenerated,
+                                         kBraveWalletIosContainerGeneratedSize,
+                                         IDR_BRAVE_WALLET_IOS_CONTAINER_HTML);
+  } else {
+    NOTREACHED() << "Failed to find page resources for:" << url.path();
+  }
+
+  for (const auto& str : brave_wallet::kLocalizedStrings) {
+    std::u16string l10n_str =
+        brave_l10n::GetLocalizedResourceUTF16String(str.id);
+    source->AddString(str.name, l10n_str);
+  }
+
+  source->AddBoolean("isAndroid", false);
+  source->AddBoolean("isIOS", true);
+
+  source->AddString("braveWalletLedgerBridgeUrl", kUntrustedLedgerURL);
+  source->AddString("braveWalletTrezorBridgeUrl", kUntrustedTrezorURL);
+  source->AddString("braveWalletNftBridgeUrl", kUntrustedNftURL);
+  source->AddString("braveWalletLineChartBridgeUrl", kUntrustedLineChartURL);
+  source->AddString("braveWalletMarketUiBridgeUrl", kUntrustedMarketURL);
+  source->AddBoolean(brave_wallet::mojom::kP3ACountTestNetworksLoadTimeKey,
+                     base::CommandLine::ForCurrentProcess()->HasSwitch(
+                         brave_wallet::mojom::kP3ACountTestNetworksSwitch));
+  source->OverrideContentSecurityPolicy(
+      network::mojom::CSPDirectiveName::FrameSrc,
+      std::string("frame-src ") + kUntrustedTrezorURL + " " +
+          kUntrustedLedgerURL + " " + kUntrustedNftURL + " " +
+          kUntrustedLineChartURL + " " + kUntrustedMarketURL + ";");
+  source->OverrideContentSecurityPolicy(
+      network::mojom::CSPDirectiveName::ImgSrc,
+      "img-src 'self' data: chrome://resources chrome://erc-token-images "
+      "chrome://image;");
```

Comment:
What's the reason we need to redefine the CSP here rather than reusing the ones defined in brave wallet UI? I see we're overriding the CSP in a few places and don't want us to accidentally miss updates on the CSP here if we make changes to the wallet UI code or vice versa.